cmd/snap/inhibit.go

```diff
@@ -195,45 +200,52 @@ func (gf *graphicalFlow) FinishInhibitionNotification(ctx context.Context) error
 	}
 
 	// finish graphical session flow
-	finishRefreshInfo := client.FinishedSnapRefreshInfo{InstanceName: gf.instanceName}
+	finishRefreshInfo := userclient.FinishedSnapRefreshInfo{InstanceName: gf.instanceName}
 	return finishRefreshNotification(ctx, &finishRefreshInfo)
 }
 
-var tryNotifyRefreshViaSnapDesktopIntegrationFlow = func(ctx context.Context, snapName string) (notified bool) {
-	// Check if Snapd-Desktop-Integration is available
-	conn, err := dbusutil.SessionBus()
+var tryNotifyRefreshViaSnapDesktopIntegrationFlow = func(ctx context.Context, cli *client.Client, instanceName string) (notified bool, err error) {
+	// check if marker interface "snap-refresh-observe" is connected
+	connOpts := client.ConnectionOptions{
+		Interface: "snap-refresh-observe",
+	}
+	connections, err := cli.Connections(&connOpts)
 	if err != nil {
-		logger.Noticef("unable to connect dbus session: %v", err)
-		return false
+		return false, err
+	}
+	if len(connections.Established) == 0 {
+		// marker interface is not connected.
+		// no snap (i.e. snapd-desktop-integration) is listening, let's fallback
+		// to normal graphical flow.
+		return false, nil
+	}
```

Comment:
the original graphical flow never really worked before, so not sure we should keep it?

Reply:
Great point, better to remove it. Thanks!

Reply:
Removed, thank you!